### ISSUE TITLE
Allow to set a custom com.jcraft.jsch.Logger for each session or Jsch instance

### ISSUE THREAD

**Issue by lukicys** [mwiede/jsch#97]
The issue is that we can run several sessions in parallel.

Current implementation makes as several problems:
- Logs of one session are lost or miss entries.
- Logs contain spurious entries (that miss from other sessions).
- we get some exceptions because of multi-threading

**Comment by mwiede** [comment]
sorry, original Jsch per se is not thread-safe

**Comment by kimmerin** [comment]
I've missed this issue when creating mine. Shall we declare this a duplicate of #127 and close this one?
